src/commons/prefetch.ts

```diff
@@ -1,16 +1,40 @@
-import { dehydrate, QueryClient } from 'react-query';
+import {
+  dehydrate,
+  FetchInfiniteQueryOptions,
+  FetchQueryOptions,
+  QueryClient,
+  QueryFunction,
+  QueryKey,
+  DehydratedState,
+} from 'react-query';
 
-export interface ResourcesType {
-  key: Array<any>;
-  fetcher: (...args: any) => void;
+interface ResourcesType<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+> {
+  key: TQueryKey;
+  fetcher: QueryFunction<TQueryFnData, TQueryKey>;
+  options?:
+    | FetchQueryOptions<TQueryFnData, TError, TData, TQueryKey>
+    | FetchInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>;
+  useInfiniteQuery?: boolean;
 }
```

Comment:
ResourcesType타입을 이렇게 지정하면, useInfinityQuery가 true인 경우만 options 타입이 FetchInfiniteQueryOptions이 되는데요
하단 코드에서는  useInfinityQuery가 ture라면  FetchInfiniteQueryOptions타입으로 추론되도록 type guard를 사용하면 되지 않을까 싶습니다!

<img width="1005" alt="image" src="https://user-images.githubusercontent.com/39763891/218103462-2077bd41-f6fb-423b-9af9-c25d1cfe1687.png">


```suggestion
type ResourcesType<
  TQueryFnData = unknown,
  TError = unknown,
  TData = TQueryFnData,
  TQueryKey extends QueryKey = QueryKey,
> = {
  key: TQueryKey;
  fetcher: QueryFunction<TQueryFnData, TQueryKey>;
} & (
  | {
      options?: FetchQueryOptions<TQueryFnData, TError, TData, TQueryKey>;
      useInfiniteQuery?: false;
    }
  | {
      options?: FetchInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>;
      useInfiniteQuery: true;
    }
);
```

Reply:
오 감사합니다! 적용할게요

Reply:
적용했습니다! 좋은 의견 감사드립니다 :)